src/escape.rs

```diff
@@ -123,14 +127,166 @@ where
         }
         let c = bytes[i];
         let escape = HTML_ESCAPE_TABLE[c as usize];
-        if escape != 0 && (secure || c != b'/') {
+        if escape != 0 {
             let escape_seq = HTML_ESCAPES[escape as usize];
             w.write_all(&bytes[mark..i])?;
             w.write_all(escape_seq.as_bytes())?;
             mark = i + 1; // all escaped characters are ASCII
         }
         i += 1;
     }
-    w.write_all(&bytes[mark..])?;
-    Ok(())
+    w.write_all(&bytes[mark..])
+}
+
+#[cfg(all(target_arch = "x86_64", feature="simd"))]
+mod simd {
+    use std::arch::x86_64::*;
+    use std::io::{self, Write};
+    use std::mem::size_of;
+
+    const VECTOR_SIZE: usize = size_of::<__m128i>();
+
+    pub(crate) fn escape_html<W: Write>(mut w: W, s: &str) -> io::Result<()> {
+        // The SIMD accelerated code uses the PSHUFB instruction, which is part
+        // of the SSSE3 instruction set. Further, we can only use this code if
+        // the buffer is at least one VECTOR_SIZE in length to prevent reading 
+        // out of bounds. If either of these conditions is not met, we fall back
+        // to scalar code.
+        if is_x86_feature_detected!("ssse3") && s.len() >= VECTOR_SIZE {
+            let bytes = s.as_bytes();
+            let mut mark = 0;
+
+            unsafe {
+                foreach_special_simd(bytes, 0, |i| {
+                    let replacement = super::HTML_ESCAPES[super::HTML_ESCAPE_TABLE[bytes[i] as usize] as usize];
+                    w.write_all(&bytes[mark..i])?;
+                    mark = i + 1; // all escaped characters are ASCII
+                    w.write_all(replacement.as_bytes())
+                })?;
+            }
+            w.write_all(&bytes[mark..])
+        } else {
+            super::escape_html_scalar(w, s)
+        }
+    }
+
+    /// Creates the lookup table for use in `compute_mask`.
+    const fn create_lookup() -> [u8; 16] {
+        let mut table = [0; 16];
+        table[(b'<' & 0x0f) as usize] = b'<';
+        table[(b'>' & 0x0f) as usize] = b'>';
+        table[(b'&' & 0x0f) as usize] = b'&';
+        table[(b'"' & 0x0f) as usize] = b'"';
+        table[0] = 0b0111_1111;
+        table
+    }
+
+    #[target_feature(enable = "ssse3")]
+    /// Computes a byte mask at given offset in the byte buffer. Its first 16 (least significant)
+    /// bits correspond to whether there is an HTML special byte (&, <, ", >) at the 16 bytes
+    /// `bytes[offset..]`. For example, the mask `(1 << 3)` states that there is an HTML byte
+    /// at `offset + 3`. It is only safe to call this function when
```

Comment:
It might be clearer to put the `debug_assert` on length here, because this function is the one for which the safety condition is critical, and it would also get coverage for the final segment. But it's ok either way.